cli/src/add_url.rs

```diff
@@ -1,16 +1,14 @@
-// File: add_url.rs
 use color_eyre::Result;
-use db::{AddUrlOutcome, PgPool};
+use db::{add_url::AddUrlOutcome, PgPool};
 use uuid::Uuid;
 
 use crate::check_auth_status;
-
 pub async fn add_url(
     pool: &PgPool,
     url: &str,
     user_id: Uuid,
     allow_existing: bool,
 ) -> Result<AddUrlOutcome> {
     check_auth_status(pool).await?;
-    db::add_url(pool, url, user_id, &allow_existing).await
+    db::add_url::add_url(pool, url, user_id, &allow_existing).await
```

Comment:
I wasn't able call `add_url` directly on line 13

I tried importing it on line 2 as `use db::{add_url, add_url::AddUrlOutcome, PgPool};`

and it gave me this error

`error[E0733]: recursion in an async fn requires boxing
  --> cli/src/add_url.rs:6:1
   |
6  | / pub async fn add_url(
7  | |     pool: &PgPool,
8  | |     url: &str,
9  | |     user_id: Uuid,
10 | |     allow_existing: bool,
11 | | ) -> Result<AddUrlOutcome> {
   | |__________________________^
   |
   = note: a recursive `async fn` call must introduce indirection such as `Box::pin` to avoid an infinitely sized future

`

Reply:
Ahh nice this is great!

I think your problem in that error message is that `add_url` was _also_ the name of this method you are defining!
So it thought you were trying to call this method recursively instead of calling the one in `db`!

So this full pathing works for me!

Reply:
oh that's silly of me 😂

Reply:
There _might_ have also been a warning there about `add_url` in your import  being 'shadowed' by this function but I'm not positive about that...

By 'shadowed' I mean that your import brought `add_url` into scope, but once you define the function you can't refer to the `db::add_url` as just `add_url` anymore because we 'shadowed' it with the local function

'Shadowing' is also the term used when you have multiple `let`s that have the same name, but may be different types!

```rust
let x: i64 = 6;
let x: String = x.to_string();
```

Reply:
That makes sense.